File: daq/slc/src/dqm/src/TimedGraph1F.cc
#include "TimedGraph1F.h"

#include "FloatArray.h"

using namespace Belle2;

TimedGraph1F::TimedGraph1F() throw() : TimedGraph1()
{
  _data = new FloatArray(0);
}

TimedGraph1F::TimedGraph1F(const TimedGraph1F& h) throw() : TimedGraph1(h)
{
  _data = new FloatArray(h._data->size());
}

TimedGraph1F::TimedGraph1F(const std::string& name, const std::string& title,
                           int nbinx, double xmin, double xmax)
throw() : TimedGraph1(name, title, nbinx, xmin, xmax)
{
  _data = new FloatArray(nbinx);
  _time_v.resize(nbinx);
}

TimedGraph1F::~TimedGraph1F() throw() {}

std::string TimedGraph1F::getDataType() const throw()
{
  return std::string("TGF");
}

File: daq/slc/extra/javalib/b2daq/src/b2daq/dqm/core/TimedGraph1F.java
package b2daq.dqm.core;

public class TimedGraph1F extends TimedGraph1 {

	public TimedGraph1F() {
		super();
		data = new FloatArray();
	}

	public TimedGraph1F(String name, String title, int nbins, double xmin, double xmax) {
		super(name, title, nbins, xmin, xmax);
		data = new FloatArray(nbins);
	}

	public String getDataType() {
		return "TGF";
	}

}
